### PR TITLE
[STORY-443] Add flag --no-letsencrypt when creating a domain

### PR DESCRIPTION
**Pull request by leo-scalingo**
- [x] Add a changelog entry in the section "To Be Released" of CHANGELOG.md

Add:

```
scalingo domains-add mydomain.example.com --no-letsencrypt
```

### CODE REVIEW

**Comment by leo-scalingo** [comment]
```
+-----------------------+---------+---------+---------------------------+
|        DOMAIN         | TLS/SSL | SUBJECT | LET'S ENCRYPT CERTIFICATE |
+-----------------------+---------+---------+---------------------------+
| one-off-2.example.com | -       |         | Disabled                  |
| one-off.example.com   | -       |         | Pending DNS               |
+-----------------------+---------+---------+---------------------------+
```

```
+----------------------------------------+--------------------------------+---------------------------+---------------------------+
|                 DOMAIN                 |            TLS/SSL             |          SUBJECT          | LET'S ENCRYPT CERTIFICATE |
+----------------------------------------+--------------------------------+---------------------------+---------------------------+
| mydomain.example.com | Valid until 2025-04-09         | /CN=*.mydomain.example.com | Created, Not in used               |
|                                        | 23:59:59 +0000 +0000           |                           |                           |
+----------------------------------------+--------------------------------+---------------------------+---------------------------+

```

```
+-------------------------+---------------+-----------------------------+---------------------------+
|         DOMAIN          |    TLS/SSL    |           SUBJECT           | LET'S ENCRYPT CERTIFICATE |
+-------------------------+---------------+-----------------------------+---------------------------+
| api.osc-st-fr1.st-sc.fr | Let's Encrypt | /CN=api.osc-st-fr1.st-sc.fr | Created                   |
+-------------------------+---------------+-----------------------------+---------------------------+
```

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,7 +2,8 @@
 
 ### To be Released
 
-* fix(cmd): addons-info working for non-db addons ([PR#1055](https://github.com/Scalingo/cli/pull/1055))
+* feat(domains): `domains-add` now supports `--no-letsencrypt` flag to completely disable Let's Encrypt certificat generation
```
Comment (curzolapierre) [inline CHANGELOG.md:5]:
```suggestion
* feat(domains): `domains-add` now supports `--no-letsencrypt` flag to completely disable Let's Encrypt certificat generation ([PR#1058](https://github.com/Scalingo/cli/pull/1058))
```

File: go.mod [hunk 2]
```diff
@@ -35,6 +35,7 @@ require (
 	dario.cat/mergo v1.0.0 // indirect
 	github.com/Microsoft/go-winio v0.6.1 // indirect
 	github.com/ProtonMail/go-crypto v1.1.0-alpha.2 // indirect
+	github.com/Scalingo/go-scalingo/v7 v7.0.0 // indirect
 	github.com/VividCortex/ewma v1.2.0 // indirect
```
Comment (curzolapierre) [inline go.mod:39]: You should run `go mod tidy` to replace v6

File: domains/list.go [hunk 3]
```diff
@@ -40,18 +40,32 @@ func List(ctx context.Context, app string) error {
...
+		tls := "-"
+		letsEncrypt := "Disabled"
+		if domain.LetsEncryptEnabled {
+			// If the domain is using Let's Encrypt (and not a custom cert), we mention it
+			if domain.LetsEncrypt {
+				tls = "Let's Encrypt"
+			}
+			// In any case we display the state of creation of the Let's Encrypt certificate
+			// So if a customer certification is used, it is still mentionned we have it
+			var ok bool
```
Comment (curzolapierre) [inline domains/list.go:53]:
```suggestion
			// So if a customer certification is used, it is still mentioned we have it
```

**Review by curzolapierre — CHANGES_REQUESTED** [review]
Overall LGTM, just a typo and import to fix

File: domains/list.go [hunk 4]
```diff
@@ -40,18 +40,32 @@ func List(ctx context.Context, app string) error {
...
+
+		tls := "-"
+		letsEncrypt := "Disabled"
+		if domain.LetsEncryptEnabled {
+			// If the domain is using Let's Encrypt (and not a custom cert), we mention it
+			if domain.LetsEncrypt {
+				tls = "Let's Encrypt"
+			}
+			// In any case we display the state of creation of the Let's Encrypt certificate
+			// So if a customer certification is used, it is still mentionned we have it
```
Comment (curzolapierre) [inline domains/list.go:52]:
```suggestion
			// So if a customer certification is used, it is still mentioned we have it
```

File: go.mod (hunk 2 above)
Comment (leo-scalingo) [reply go.mod:39]: Done

**Review by curzolapierre — APPROVED** [review]

**Pull request merged**